speeduino/canBroadcast.ino

```diff
@@ -0,0 +1,117 @@
+/*
+Speeduino - Simple engine management for the Arduino Mega 2560 platform
+Copyright (C) Josh Stewart
+A full copy of the license may be found in the projects root directory
+*/
+
+/*
+This is for handling the data broadcasted to various CAN dashes and instrument clusters.
+*/
+#if defined(NATIVE_CAN_AVAILABLE)
+#include "globals.h"
+
+void sendBMWCluster()
+{
+  DashMessage(CAN_BMW_DME1);
+  Can0.write(outMsg);
+  DashMessage(CAN_BMW_DME2);
+  Can0.write(outMsg);
+  DashMessage(CAN_BMW_DME4);
+  Can0.write(outMsg);
+}
+
+void sendVAGCluster()
+{
+  DashMessage(CAN_VAG_RPM);
+  Can0.write(outMsg);
+  DashMessage(CAN_VAG_VSS);
+  Can0.write(outMsg);
+}
+
+// switch case for gathering all data to message based on CAN Id.
+void DashMessage(uint16_t DashMessageID)
+{
+  switch (DashMessageID)
+  {
+    case CAN_BMW_DME1:
+      uint16_t temp_RPM;
+      temp_RPM =  currentStatus.RPM * 6.4; //RPM conversion
```

Comment:
Use of floats must be avoided at the moment, it is on lines 38 and 100.

Reply:
> Use of floats must be avoided at the moment, it is on lines 38 and 100.

Fixed